routes/shopItem.js

```diff
@@ -7,10 +7,18 @@ shopItemRouter = express.Router();
 
 shopItemRouter.post("/", (req, res, next) => {
     try {
-        id++;
-        req.body.id = id;
-        fs.writeFileSync(resolve("./database.json"), JSON.stringify(req.body) + "\n", { flag: "a+" });
-        res.json(`${req.method} success`);
+        if (typeof(req.body.name) === "string" && typeof(req.body.isCompleted === "boolean")) {
+            if ("isCompleted" in req.body === false) {
+                req.body.isCompleted = false;
+            };
```

Comment:
This should be done before the `if` in the previous (`if`) line otherwise when `isCompleted` is not in the body, the condition of that if will be false since typeof(req.body.isCompleted) would be `undefined`.

Therefore adding isCompleted with `false` value (when missing) should come before the type check.